# Upstream-Status: inapropriate
# Fix up for building the girs only:
#  - remove depencies on g-ir-compiler (we build that a standalone tools)
#  - remove various pkg-config crud, this belongs into the configure script
#    (for which we have configure-introspection.patch)
#  - replace references to libgirepository-1.0.la with girepository-1.0.so,
#    since we also built this earlier and are using pkg-config to locate it
Index: gobject-introspection-1.34.2/Makefile-gir.am
===================================================================
--- gobject-introspection-1.34.2.orig/Makefile-gir.am	2013-03-27 11:43:59.253349026 +0000
+++ gobject-introspection-1.34.2/Makefile-gir.am	2013-03-27 11:47:43.632518431 +0000
@@ -25,19 +25,6 @@
 CLEANFILES += gir/cairo-1.0.gir
 EXTRA_DIST += gir/cairo-1.0.gir.in
 
-# Doesn't seem to work to do gir/%.typelib =(
-gir/cairo-1.0.typelib: g-ir-compiler
-gir/DBus-1.0.typelib: g-ir-compiler
-gir/DBusGLib-1.0.typelib: g-ir-compiler
-gir/fontconfig-2.0.typelib: g-ir-compiler
-gir/freetype2-2.0.typelib: g-ir-compiler
-gir/GL-1.0.typelib: g-ir-compiler
-gir/libxml2-2.0.typelib: g-ir-compiler
-gir/xft-2.0.typelib: g-ir-compiler
-gir/xlib-2.0.typelib: g-ir-compiler
-gir/xfixes-4.0.typelib: g-ir-compiler
-gir/xrandr-1.3.typelib: g-ir-compiler
-
 # glib with source (to enable documentation)
 if WITH_GLIBSRC
      GLib_2_0_gir_DOCSRC = $(GLIBSRC)/glib/*.c
@@ -53,9 +40,6 @@
 endif
 
 # glib
-GLIB_INCLUDEDIR=$(shell pkg-config --variable=includedir glib-2.0)/glib-2.0
-GLIB_LIBDIR=$(shell pkg-config --variable=libdir glib-2.0)
-
 if OS_WIN32
 GLIB_LIBRARY=libglib-2.0-0
 else
@@ -89,14 +73,9 @@
 
 BUILT_GIRSOURCES += GLib-2.0.gir
 
-GLib-2.0.gir: g-ir-scanner g-ir-compiler$(EXEEXT)
-
 gir/DBusGLib-1.0.typelib: GObject-2.0.gir
 
 # gobject
-GOBJECT_INCLUDEDIR=$(shell pkg-config --variable=includedir gobject-2.0)/glib-2.0
-GOBJECT_LIBDIR=$(shell pkg-config --variable=libdir gobject-2.0)
-
 if OS_WIN32
 GOBJECT_LIBRARY=libgobject-2.0-0
 else
@@ -126,9 +105,6 @@
 BUILT_GIRSOURCES += GObject-2.0.gir
 
 # gmodule
-GMODULE_INCLUDEDIR=$(shell pkg-config --variable=includedir gmodule-2.0)/glib-2.0
-GMODULE_LIBDIR=$(shell pkg-config --variable=libdir gmodule-2.0)
-
 if OS_WIN32
 GMODULE_LIBRARY=libgmodule-2.0-0
 else
@@ -155,9 +131,6 @@
 BUILT_GIRSOURCES += GModule-2.0.gir
 
 # gio
-GIO_INCLUDEDIR=$(shell pkg-config --variable=includedir gio-2.0)/glib-2.0
-GIO_LIBDIR=$(shell pkg-config --variable=libdir gio-2.0)
-
 if OS_WIN32
 GIO_LIBRARY=libgio-2.0-0
 else
@@ -165,13 +138,10 @@
 endif
 
 if HAVE_GIO_UNIX
-GIO_UNIX_HDRS=$(shell pkg-config --variable=includedir gio-unix-2.0)/gio-unix-2.0/gio/*.h
-GIO_UNIX_PACKAGES = gio-unix-2.0
 else
 GIO_UNIX_HDRS=
 GIO_UNIX_PACKAGES =
 endif
-GIO_CINCLUDES=$(patsubst %,--c-include='%',$(shell echo $(GIO_UNIX_HDRS)))
 
 Gio-2.0.gir: GObject-2.0.gir
 
@@ -200,9 +170,9 @@
 EXTRA_DIST += gir/glib-2.0.c gir/gobject-2.0.c gir/gio-2.0.c
 
 # girepository
-GIRepository-2.0.gir: GObject-2.0.gir libgirepository-1.0.la
+GIRepository-2.0.gir: GObject-2.0.gir
 
-GIRepository_2_0_gir_LIBS = libgirepository-1.0.la
+GIRepository_2_0_gir_LIBS = girepository-1.0
 GIRepository_2_0_gir_SCANNERFLAGS = \
 	--warn-all \
         --identifier-prefix=GI \
@@ -213,7 +183,7 @@
 GIRepository_2_0_gir_INCLUDES = GObject-2.0
 GIRepository_2_0_gir_EXPORT_PACKAGES = gobject-introspection-1.0
 GIRepository_2_0_gir_CFLAGS = \
-    -I$(top_srcdir)/girepository -I$(top_srcdir) \
+    $(GIR_CFLAGS) -I$(top_srcdir) \
     -DGI_COMPILATION
 GIRepository_2_0_gir_FILES = \
         $(top_srcdir)/girepository/girepository.h   \
